dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -911,8 +912,8 @@ private Workflow createOrGetWorkflow(Class workflowType, String repository, User
         // Check that the workflow is writable
         checkWritability(workflowToUpdate);
 
-        if (user != null) {
-            workflowToUpdate.getUsers().add(user);
+        if (user.isPresent()) {
+            workflowToUpdate.getUsers().add(user.get());
```

Comment:
Strong opinion: when we use `Optional`, we should try to use patterns that reduce the failure modes that are similar to NPEs (calling `get` before `isPresent`, etc). because otherwise it's just more words and the same problems:
```
user.ifPresent(workflowToUpdate.getUsers()::add);
```

Reply:
Interesting, my initial attempt was `user.ifPresent(u -> workflowToUpdate.getUsers().add(u))`, but the compiler nixed it because `workflowToUpdate` is not effectively final. I didn't realize your way would work, which it does (and it's more compact).